usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedFeedInfoTest.java

```diff
@@ -0,0 +1,248 @@
+
+        //noinspection unchecked
+        when(mockBuilder.build()).thenReturn(mockGenericObject);
+
+        final ProcessParsedFeedInfo underTest = new ProcessParsedFeedInfo(mockResultRepo, mockGtfsDataRepo,
+                mockBuilder);
+
+        when(mockParsedFeedInfo.get(FEED_PUBLISHER_NAME)).thenReturn(FEED_PUBLISHER_NAME);
+        when(mockParsedFeedInfo.get(FEED_PUBLISHER_URL)).thenReturn(FEED_PUBLISHER_URL);
+        when(mockParsedFeedInfo.get(FEED_LANG)).thenReturn(FEED_LANG);
+        when(mockParsedFeedInfo.get(FEED_START_DATE)).thenReturn(START_DATE);
+        when(mockParsedFeedInfo.get(FEED_END_DATE)).thenReturn(END_DATE);
+        when(mockParsedFeedInfo.get(FEED_VERSION)).thenReturn(FEED_VERSION);
+        when(mockParsedFeedInfo.get(FEED_CONTACT_EMAIL)).thenReturn(FEED_CONTACT_EMAIL);
+        when(mockParsedFeedInfo.get(FEED_CONTACT_URL)).thenReturn(FEED_CONTACT_URL);
+
+        underTest.execute(mockParsedFeedInfo);
+
+        verify(mockParsedFeedInfo, times(1)).get(ArgumentMatchers.eq(FEED_PUBLISHER_NAME));
+        verify(mockParsedFeedInfo, times(1)).get(ArgumentMatchers.eq(FEED_PUBLISHER_URL));
+        verify(mockParsedFeedInfo, times(1)).get(ArgumentMatchers.eq(FEED_LANG));
+        verify(mockParsedFeedInfo, times(1)).get(ArgumentMatchers.eq(FEED_START_DATE));
+        verify(mockParsedFeedInfo, times(1)).get(ArgumentMatchers.eq(FEED_END_DATE));
+        verify(mockParsedFeedInfo, times(1)).get(ArgumentMatchers.eq(FEED_VERSION));
+        verify(mockParsedFeedInfo, times(1)).get(ArgumentMatchers.eq(FEED_CONTACT_EMAIL));
+        verify(mockParsedFeedInfo, times(1)).get(ArgumentMatchers.eq(FEED_CONTACT_URL));
+
+        verify(mockBuilder, times(1))
+                .feedPublisherName(ArgumentMatchers.eq(FEED_PUBLISHER_NAME));
+        verify(mockBuilder, times(1)).feedPublisherUrl(ArgumentMatchers.eq(FEED_PUBLISHER_URL));
+        verify(mockBuilder, times(1)).feedLang(ArgumentMatchers.eq(FEED_LANG));
+        verify(mockBuilder, times(1)).feedStartDate(ArgumentMatchers.eq(START_DATE));
+        verify(mockBuilder, times(1)).feedEndDate(ArgumentMatchers.eq(END_DATE));
+        verify(mockBuilder, times(1)).feedVersion(ArgumentMatchers.eq(FEED_VERSION));
+        verify(mockBuilder, times(1)).feedContactEmail(ArgumentMatchers.eq(FEED_CONTACT_EMAIL));
+        verify(mockBuilder, times(1)).feedContactUrl(ArgumentMatchers.eq(FEED_CONTACT_URL));
+        verify(mockBuilder, times(1)).build();
+
+        verify(mockGenericObject, times(1)).isSuccess();
+        //noinspection ResultOfMethodCallIgnored
+        verify(mockGenericObject, times(1)).getData();
+
+        verify(mockResultRepo, times(1)).addNotice(isA(Notice.class));
+        verifyNoMoreInteractions(mockParsedFeedInfo, mockGtfsDataRepo, mockBuilder, mockResultRepo, mockGenericObject);
+    }
+
+    @Test
+    void duplicateFeedInfoShouldAddNoticeToResultRepoAndShouldNotBeAddedToGtfsDataRepo() {
```

Comment:
Same remark as the other one, could testing at this level could be simplified to repository add rejection, no matter the cause?

Reply:
No it can't. The duplication notice is generated inside the use case, hence needs testing at that level.